lib/matplotlib/offsetbox.py

```diff
@@ -1668,14 +1668,21 @@ def on_pick(self, evt):
             self.save_offset()
 
     def on_release(self, event):
-        if self.got_artist:
+        if self._check_still_parented() and self.got_artist:
             self.finalize_offset()
             self.got_artist = False
             self.canvas.mpl_disconnect(self._c1)
 
             if self._use_blit:
                 self.ref_artist.set_animated(False)
 
+    def _check_still_parented(self):
+        if self.ref_artist.axes is None:
```

Comment:
should probably check `self.ref_artist.figure` too incase there are draggable things at the figure level.

Reply:
fixed